### PR TITLE
Update turn-debug-logging-on.md

### PR DESCRIPTION
**Pull request by tannazie**
- moved section about log files to the bottom
 - added a note that talked about the relationship between Desktop checkbox, env var, and engine setting

### CODE REVIEW

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/71af97ccbf6dd0d6962f6e5a0e94b842a4841496/index.html)

File: docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md [hunk 1]
```diff
@@ -95,6 +93,8 @@ engines:
 
 Save the file and relaunch Nuke in the shot step environment. You will now see the debug output in the script editor window. 
 
+{% include info title="Note" content="If debug logging is enabled via any one of the Desktop checkbox, environment variable, or engine configuration setting, debug logging will be output. Additionally, each of these three is modified independently of the others: the checkbox value is a QSetting, and it persists across sessions of Desktop via QSetting, totally independent of the engine setting or env var. This means that while the Desktop checkbox may be unchecked, debug logging might still be enabled by way of one of the other methods." %}
```
Comment (pscadding) [inline docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md]:
```suggestion
{% include info title="Note" content="If debug logging is enabled via any one of the Desktop checkbox, environment variable, or engine configuration setting, debug logging will be output. Additionally, each of these three is modified independently of the others: the checkbox state is stored persistently across sessions via the use of `QSettings`, totally independent of the engine setting or environment variable. This means that while the Desktop checkbox may be unchecked, debug logging might still be enabled by way of one of the other methods." %}
```

Comment (pscadding) [reply docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md]: `QSetting` isn't a thing, but `QSettings` is, and it doesn't technically return a `QSetting` value, I think it returns a `dict` with `PySide` and a `QVariant` with `PyQt` (but we don't use `PyQt` with Desktop so kinda irrelevant 😛)

Do we need to mention again that the Desktop checkbox is stored persistently, we already say that at the top? If you think important to remind them, then perhaps keep it simpler and not bother mentions the details of `QSettings` at all, and maybe bump that bit to the top if you think that's still needed.
Other than that, I think this note box is good info!

File: docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md [hunk 2]
```diff
@@ -163,3 +163,7 @@ If you’re running the tank command and want to see debug output in your termin
     You are currently running version v0.15.18 of the Shotgun Pipeline Toolkit
     No need to update the Toolkit Core API at this time!
     DEBUG [10:11:39 981.74405098]: Exiting with exit code None
+
+## Examining debug outputs in log files
```
Comment (pscadding) [inline docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md]: I can see why you've moved this down here, I agree I don't think it was in the right place before.
Just a suggestion and feel free to ignore, but I think probably 90% of people are just going to go with the Desktop checkbox, and probably not read the whole of what is quite a quite lengthy article.

Maybe we could move this up to just below the Desktop checkbox section?
And then sub group the rest of the methods as other options or somthing.

Or maybe create a TOC at the top of the doc?

Comment (tannazie) [reply docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md]: I'll move it to the top, and reword it to be a quick preface to the rest of the content

Comment (tannazie) [reply docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md]: Moved it up to the very top as a note. This was a good moment to throw in an intro sentence for the whole doc, too. Good call!

File: docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md (hunk 1 above)
Comment (tannazie) [reply docs/en/quick-answers/troubleshooting/turn-debug-logging-on.md]: omitted the QSettings details, cleaned up the wording, Desktop -> Shotgun Desktop.

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/979c69ce2004d26caf9c612182a6b5e0b507a5b5/index.html)

**Comment by toolkitty** [comment]
[Documentation Preview](http://sg-devdocs.s3-website-us-east-1.amazonaws.com/tk-doc-generator/92388ba99ab1136b600e278b6289bcaff0e0e175/index.html)
